### PR TITLE
Propagete _tag to static side

### PR DESCRIPTION
**Pull request by mikearnaldi**
we might also want to extend variants such as Schema.TaggedError etc

### CODE REVIEW

File: packages/effect/src/Data.ts [hunk 1]
```diff
@@ -441,11 +441,15 @@ export const Error: new<A extends Record<string, any> = {}>(
...
-) => Cause.YieldableError & { readonly _tag: Tag } & Readonly<A> => {
+export const TaggedError = <Tag extends string>(tag: Tag): {
+  readonly _tag: Tag
+  new<A extends Record<string, any> = {}>(
+    args: Types.Equals<A, {}> extends true ? void
+      : { readonly [P in keyof A as P extends "_tag" ? never : P]: A[P] }
+  ): Cause.YieldableError & { readonly _tag: Tag } & Readonly<A>
+} => {
   class Base extends Error<{}> {
+    static _tag = tag
```
Comment (patroza) [inline packages/effect/src/Data.ts:452]: readonly ?

Comment (mikearnaldi) [reply packages/effect/src/Data.ts:452]: type is not exposed, it's the internal impl

Comment (patroza) [reply packages/effect/src/Data.ts:452]: I know, but look below: `readonly _tag = tag`
anyway it's a non issue as you said.

**Comment by datner** [comment]
Any updates on this?

**Pull request closed without merging**